src/main/java/gregtech/common/mui/widget/GTTextFieldWidget.java

```diff
@@ -0,0 +1,259 @@
+package gregtech.common.mui.widget;
+
+import gregtech.api.util.GTLog;
+
+import com.cleanroommc.modularui.api.drawable.IKey;
+import com.cleanroommc.modularui.api.value.IStringValue;
+import com.cleanroommc.modularui.screen.RichTooltip;
+import com.cleanroommc.modularui.screen.viewport.ModularGuiContext;
+import com.cleanroommc.modularui.utils.MathUtils;
+import com.cleanroommc.modularui.utils.ParseResult;
+import com.cleanroommc.modularui.value.StringValue;
+import com.cleanroommc.modularui.value.sync.SyncHandler;
+import com.cleanroommc.modularui.value.sync.ValueSyncHandler;
+import com.cleanroommc.modularui.widgets.textfield.BaseTextFieldWidget;
+import com.cleanroommc.modularui.widgets.textfield.TextFieldHandler;
+import com.cleanroommc.modularui.widgets.textfield.TextFieldRenderer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.*;
+import java.text.ParsePosition;
+import java.util.function.DoubleUnaryOperator;
+import java.util.function.Function;
+import java.util.function.IntUnaryOperator;
+import java.util.function.LongUnaryOperator;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+public class GTTextFieldWidget extends BaseTextFieldWidget<GTTextFieldWidget> {
```

Comment:
A decent amount of this class is unused. Do we need to keep any of it?

Reply:
the methods are there to match mui2's TextFieldWidget. I could extend TextFieldWidget and remove the redundant methods, but then builder calls become more complicated due to methods returning TextFieldWidget instead of GTTextFieldWidget